### DISCUSSION TITLE
Naming / new labeling for ARM-related stuff

### DISCUSSION THREAD

**Discussion by carlocaione** [zephyrproject-rtos/zephyr#28353] [General] [answered]
Currently Zephyr only has an `area: ARM` label for all the ARM-related PRs. I think we should introduce a new `area: ARM64` label for anything related to 64-bit ARM.

This actually sparkled a discussion on IRC about the general naming since on the ARM world we have at least three variants: ARM (32-bit ARMv7), AArch64 (64-bit ARMv8 AArch64) and AArch32 (32-bit ARMv8 AArch32).

**Comment by carlocaione** [comment]
tagging some ARM interested people: @ioannisg @stephanosio @andrewboie @sandeepbrcm @Abhishek-brcm  @galak @MaureenHelm

**Comment by ioannisg** [comment, marked as answer]
> I think we should introduce a new area: ARM64 label for anything related to 64-bit ARM.

Thanks @carlocaione 
For x86 architecture we have `area: X86` and `area: X86_64`
So, would it be better to have `area: ARM` and `area: ARM_64` ?

This would mean that
- area: ARM will be for Cortex-M, Cortex-R, but also for 32-bit Cortex-A, which, currently is not supported
- area: ARM_64 will be for Cortex-A AARCH64

This approach is backwards compatible, since the ARM label is kept, but `area: ARM` still contains several different architecture variants, which we would ideally want to split in different areas.

An alternative solution is to use exact names for all architectures, for better filtering:
- `area: ARM Cortex-A`, for both AARC32 (not supported now) and AARCH64
- `area: ARM Cortex-R`
- `area: ARM Cortex-M`

This solution would deprecate `area: ARM` label, though, and I am not sure how this will affect searching over lots of closed issues/PRs with the ARM label.

**Reply by carlocaione** [reply]
I'm perfectly fine with that. `area: ARM_64` LGTM.
